controllers/drplacementcontrol_controller.go

```diff
@@ -0,0 +1,476 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"strings"
+	"time"
+
+	argov1alpha1 "github.com/argoproj/argo-cd/v3/pkg/apis/application/v1alpha1"
+	ramenv1alpha1 "github.com/ramendr/ramen/api/v1alpha1"
+	multiclusterv1alpha1 "github.com/red-hat-storage/odf-multicluster-orchestrator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	placementv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+	appsubv1 "open-cluster-management.io/multicloud-operators-subscription/pkg/apis/apps/v1"
+	appv1beta1 "sigs.k8s.io/application/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	labelAppType           = "app.kubernetes.io/type"
+	labelDRPolicy          = "ramendr.openshift.io/dr-policy"
+	placementLabel         = "cluster.open-cluster-management.io/placement"
+	subscriptionKind       = "Subscription"
+	drpcKind               = "DRPlacementControl"
+	placementKind          = "Placement"
+	requeueAfterSeconds    = 30 * time.Second
+	subscriptionAnnotation = "apps.open-cluster-management.io/subscriptions"
+	applicationSetKind     = "ApplicationSet"
+	ramenAPIVersion        = "ramendr.openshift.io/v1alpha1"
```

Comment:
nit: Let's have it sorted in a way that similar variables are together for eg kind, label, annotations and so on.